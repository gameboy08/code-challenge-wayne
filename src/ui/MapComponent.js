import React from "react";
import { compose, withProps } from "recompose";
import {
  withScriptjs,
  GoogleMap,
  Marker,
  withGoogleMap
} from "react-google-maps";

const MapComponent = compose(
  withProps({
    googleMapURL:
      "https://maps.googleapis.com/maps/api/js?key=AIzaSyCHudbqXHhHW-wtM_N5wWEOwWbLH3vVm2g&v=3.exp&libraries=geometry,drawing,places",
    loadingElement: <div style={{ height: `100%` }} />,
    containerElement: <div style={{ height: `400px` }} />,
    mapElement: <div style={{ height: `100%` }} />
  }),
  withScriptjs,
  withGoogleMap
)(({ latitude, longitude }) => {
  return (
    <GoogleMap
      defaultZoom={8}
      center={{ lat: latitude, lng: longitude }}
    >
      <Marker position={{ lat: latitude, lng: longitude }} />
    </GoogleMap>
  );
});

export default MapComponent;
