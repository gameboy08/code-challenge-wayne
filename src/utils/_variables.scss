$border: 1px solid #ebebeb;
$light: #F4F8F9;