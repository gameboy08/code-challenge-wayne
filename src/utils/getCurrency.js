// import cc from "currency-codes";
import n from "country-js";

export default function getCurrency(countryCode) {
    if(n.search(countryCode).length > 0) {
        var currency = n.search(countryCode)[0].currency
        return {
            ...currency
        }
    } else {
        return null
    }
}