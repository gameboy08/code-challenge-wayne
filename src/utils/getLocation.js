export default function getLocation(
  latitude,
  longitude,
  latitudeDirection,
  longitudeDirection
) {
  let lat,
    lng = null;

  switch (latitudeDirection.toUpperCase()) {
    case "S":
      lat = -latitude;
      break;
    case "N":
      lat = latitude;
      break;
    default:
      break;
  }
  switch (longitudeDirection.toUpperCase()) {
    case "W":
      lng = -longitude;
      break;
    case "E":
      lng = longitude;
      break;
    default:
      break;
  }
  return {
    lat,
    lng
  };
}
