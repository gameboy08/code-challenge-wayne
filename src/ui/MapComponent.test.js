import React from "react";
import { shallow } from "enzyme";
import MapComponent from "./MapComponent";
import { shallowToJson } from "enzyme-to-json";
import { airport } from "../utils/mockData";

const props = {
  lat: -airport.location.latitude,
  lng: -airport.location.longitude
};
it("MapComponent component", () => {

  const wrapped = shallow(
    <MapComponent latitude={props.lat} longitude={props.lng} />
  );
  let tree = shallowToJson(wrapped);
  expect(tree).toMatchSnapshot();
});
