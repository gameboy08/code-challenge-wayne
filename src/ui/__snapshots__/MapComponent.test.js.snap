// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<withScriptjs(withGoogleMap(Component))
  containerElement={
    <div
      style={
        Object {
          "height": "400px",
        }
      }
    />
  }
  googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyCHudbqXHhHW-wtM_N5wWEOwWbLH3vVm2g&v=3.exp&libraries=geometry,drawing,places"
  latitude={-17.25}
  loadingElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
  longitude={-145.3}
  mapElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
/>
`;

exports[`MapComponent component 1`] = `
<withScriptjs(withGoogleMap(Component))
  containerElement={
    <div
      style={
        Object {
          "height": "400px",
        }
      }
    />
  }
  googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyCHudbqXHhHW-wtM_N5wWEOwWbLH3vVm2g&v=3.exp&libraries=geometry,drawing,places"
  latitude={-17.25}
  loadingElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
  longitude={-145.3}
  mapElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
/>
`;
